service/src/amqp.ts

```diff
@@ -592,11 +589,7 @@ export class Amqp extends EventEmitter implements Client.Transport {
     if ((!this._accessToken) || this.isAccessTokenCloseToExpiry(this._accessToken)) {
       this._accessToken = await this._config.tokenCredential.getToken(this._iotHubPublicScope) as any;
     }
-    if (this._accessToken) {
-      return this._bearerTokenPrefix + this._accessToken.token;
-    } else {
-      return null;
-    }
+    return this._bearerTokenPrefix + this._accessToken.token;
```

Comment:
is this._accessToken guaranteed to exist?

Reply:
No, but the promise would get rejected if it doesn't exist

Reply:
Follow up based on a discussion with Daniel Rodriguez who works on `@azure/identity`:
It is safe to assume that for implementations of `TokenCredential` from `@azure/identity`, `getToken()` will always throw rather than returning null. But for custom implementations of `TokenCredential`, this may not be the case.

Based on this, I will change it back to what it was before.

Reply:
Addressed this